### PR TITLE
[56] Implement DB solution based on investigation results

### PR DESCRIPTION
**Pull request by khasanrashidov**
### Where

[NL-56 Implement DB solution based on investigation results](https://jjbcapital.atlassian.net/browse/NL-56)

### What

Provide a description of the goal of these changes

### How

Implemented business models according to the Db structure (schema).

### Affected Area

Auctionify.Core
Auctionify.Infrastructure

### PR Checklist

_Build_

- [x] Build Succeeded

### CODE REVIEW

**Comment by khasanrashidov** [comment]
image_2023-10-05_21-44-46

File: src/Auctionify.Core/Entities/Rate.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+	{
+		public int RecieverId { get; set; }
+
+		public virtual User Reciever { get; set; }
+
+		public int SenderId { get; set; }
+
+		public virtual User Sender { get; set; }
+
+		public uint RatingValue { get; set; }
```
Comment (TsvirkoNikita) [inline src/Auctionify.Core/Entities/Rate.cs:15]: What is the matter to use uint type here? It maps to bigint in sql, which more than just int. Looks like byte it's more suitable here

File: src/Auctionify.Infrastructure/Data/Config/RateConfiguration.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+			builder.HasOne(r => r.Sender)
+				.WithMany(r => r.SenderRates)
+				.IsRequired(true)
+				.OnDelete(DeleteBehavior.NoAction);
+
+			builder.HasOne(r => r.Lot)
+				.WithOne(l => l.Rate)
+				.IsRequired(false);
+
+			builder.Property(r => r.RatingValue).HasMaxLength(5).IsRequired(true);
```
Comment (TsvirkoNikita) [inline src/Auctionify.Infrastructure/Data/Config/RateConfiguration.cs:26]: HasMaxLength is using to specify a maximum length, not value. You need to change the type for RatingValue

**Review by TsvirkoNikita — CHANGES_REQUESTED** [review]

File: src/Auctionify.Core/Entities/Rate.cs [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+		public virtual User Sender { get; set; }
+
+		public byte RatingValue { get; set; }
+
+		public string Comment { get; set; }
+
+		public virtual Lot Lot { get; set; }
+
+		public int LoId { get; set; }
```
Comment (TsvirkoNikita) [inline src/Auctionify.Core/Entities/Rate.cs:21]: Fix the typo

**Review by TsvirkoNikita — CHANGES_REQUESTED** [review]

**Review by TsvirkoNikita — APPROVED** [review]

**Pull request merged**